src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -726,7 +726,12 @@ export class HubConnection {
             {
                 let nextPing = this._nextKeepAlive - new Date().getTime();
                 if (nextPing < 0) {
-                    nextPing = 0;
+                    if (this._connectionState === HubConnectionState.Connected) {
+                        this._sendMessage(this._cachedPingMessage).catch((e) => {
```

Comment:
This change is different from what you showed me. The check is now inside the `if (this._pingServerHandle === undefined)` condition, but isn't the problem that we set a timer but it isn't being fired? So this new code will never be hit in the problematic case?